src/targets/elixir/httpoison.js

```diff
@@ -0,0 +1,134 @@
+/**
+ * @description
+ * HTTPoison Elixir HTTP client code snippet generator.
+ *
+ * @author
+ * @danirukun
+ *
+ * For any questions or issues regarding the generated code snippet,
+ * please open an issue mentioning the author.
+ */
+
+'use strict'
+
+const CodeBuilder = require('../../helpers/code-builder')
+
+/**
+   Converts an ES6 list of objects to an Elixir KW list
+*/
+const jsListToKwList = (jsList) => {
+  if (jsList.length === 0) {
+    return '[]'
+  } else {
+    const kwList = jsList
+      .map(obj => `{"${obj.name}", "${obj.value}"}`)
+      .join(', ')
+    return `[${kwList}]`
+  }
+}
+
+/**
+   Converts a list containing cookies objects into an
+   HTTPoison `cookie` parameter argument.
+*/
+const jsCookiesToExBin = (jsCookiesList) => {
+  if (jsCookiesList.length === 0) {
+    return '""'
+  } else {
+    const exBinary = jsCookiesList
+      .map(cookie => `${cookie.name}=${cookie.value}`)
+      .join('; ')
+    return `"${exBinary}"`
+  }
+}
+
+/**
+   Converts a list of params to a Poison multipart form payload.
+*/
+const jsMultiPartParamsToPoisonPayload = (jsMultiParams) => {
+  const [params] = jsMultiParams
+  const isFileUpload = 'fileName' in params
+  let multiPart
+
+  if (isFileUpload) {
+    const { fileName, name } = params
+    const formData = [
+      { name: 'name', value: name },
+      { name: 'filename', value: fileName }]
+    const formDataKwList = jsListToKwList(formData)
+    const headers = jsListToKwList([{ name: 'content-type', value: 'multipart/form-data' }])
+
+    multiPart = `[{:file, "${fileName}", {"form-data", ${formDataKwList}}, ${headers}}]`
+  } else {
+    multiPart = jsListToKwList(jsMultiParams)
+  }
+
+  return `{:multipart, ${multiPart}}`
+}
+
+const isJson = (mimeType) => {
+  return mimeType === 'application/json'
```

Comment:
what about accounting for other mime types like `'application/json', 'application/x-json', 'text/json', 'text/x-json'`, etc?

Reply:
Added missing MIME types